### DISCUSSION TITLE
Simple(r) solution for playlists sync/backup

### DISCUSSION THREAD

**Discussion by bilogic** [navidrome/navidrome#3107] [Ideas]
https://github.com/navidrome/navidrome/issues/2035
https://github.com/navidrome/navidrome/issues/753

@deluan 

Continuing from the 2 issues above, it can be quite horrifying to lose my updated playlists in Navidrome.
The problem seems to be, people just need the playlist, but there is no automated way to get it.

My suggestion is:

1. Automatically export the playlists as `m3u` files into the music folder daily/hourly, or best, whenever changes occur, I'm suggesting `/.navidrome/exported-playlists/`
2. Navidrome will never import from this folder
3. For those who wish to sync another player with changes from Navidrome, just copy from this folder
4. Likewise for those who wish to perform backups
5. When we need to import again, just move these playlists somewhere else in the music folder and scan

We should not care if the user decides to unilaterally restructure their folders, that is an impossible problem to solve. If they do that, then onus is on them to open up the `m3u` files and update the paths. Likewise, if they map their folders differently to different machines, that would be on them to update the `m3u` files, or map the folders exactly the same.

2 way syncing is hard. but if it is that important, a simple way is to import and replace any playlist with the same name, we could be very specific and use the folder `/.navidrome/playlists-to-import`, the `m3u` files are removed once imported.

**Comment by metalheim** [comment]
The playlists are in the database, so for pure backup purposes, It's best to just Backup the database and youre done. Database can also be easily re-imported (just replace your db from Backup).

To Export to m3u, there is a command line option
https://github.com/navidrome/navidrome/issues/1914#issuecomment-1362012971
 and API options, but only for one-by-one Export, so youd need to run a Script to Export all playlists

**Reply by bilogic** [reply]
![image](https://github.com/navidrome/navidrome/assets/946010/54809a7a-af7b-4e60-bcad-cfbddd7a6fb3)

1. I'm sure the database contains other info right? I only want the playlists, can a restore be achieved without breaking Navidrome? And would the other use case of syncing be addressed?
2. CLI is pretty inaccessible as I'm running Navidrome as a Home Assistant add-on, but let me give it a shot. (And imagine the effort one would need to keep the backup script up to date with the playlists UUID) 

UPDATE: yup, no docker

My conclusion: of course there are options, but they are not precise enough to perform playlists backup and sync (properly).

**Comment by ERisEPR** [comment]
I agree that a all-playlist only backup would be great. In the past I've had to export 250 playlists one by one about 4 times. I have had no luck using the same database and i take the opportunity to recreate the database when switching servers or docker instances to keep things running smoothly. But the playlist export makes things very tedious. So i am +1ing this !  Thank you!
